crates/turbo-tasks-macros/src/derive/task_input_macro.rs

```diff
@@ -104,13 +104,15 @@ pub fn derive_task_input(input: TokenStream) -> TokenStream {
                 .map(Literal::usize_unsuffixed)
                 .collect();
 
+            let ident_str = ident.to_string();
+
             (
                 quote! {
                     match value {
                         turbo_tasks::ConcreteTaskInput::List(value) => {
                             let mut #inputs_list_ident = value.iter();
 
-                            let discriminant = #inputs_list_ident.next().ok_or_else(|| anyhow::anyhow!(concat!("missing discriminant for ", stringify!(#ident))))?;
+                            let discriminant = turbo_tasks::macro_helpers::ok_or_else_for_missing_element(#inputs_list_ident.next(), #ident_str)?;
```

Comment:
Maybe just turn this into a `let-else`? (same for the other 2)
not the biggest fan of that helper function unless that single thing has a major impact on compile times